lib_static/open_project/authentication/strategies/warden/jwt_oidc.rb

```diff
@@ -30,7 +30,7 @@ def authenticate!
                                            error_description: "Requires scope #{scope} to access this resource."
                 end
 
-                user = provider.user_auth_provider_links.find_by(external_id: payload["sub"])&.user
+                user = provider.user_auth_provider_links.find_by(external_id: payload["sub"])&.principal
```

Comment:
:yellow_circle: Similar to the comment in `OmniauthService`: This piece of code is not expected to authenticate groups ever. Groups are not expected/allowed to authenticate in open project, so I think we should make sure that they can't.